src/union_find_struct.jl

```diff
@@ -0,0 +1,22 @@
+struct UnionFind
+    rep::Dict{Int, Int} # maps each element to its representative
+end
+
+UnionFind(elems::Vector{Int}) = UnionFind(Dict{Int, Int}(x => x for x in elems))
+
+function get_rep(uf::UnionFind, x::Int)
+    while uf.rep[x] != x
```

Comment:
I wonder if this reduction of uf.rep[x] by 1 makes any difference at all:

```Julia
while (r = uf.ref[x]) != x
    x = uf.ref[x] = uf.ref[r]
end
```

Or allocating r is worse than querying the Dict one more time.

A shame this wouldn't work: x = r = uf.ref[r]